tests/mdns_test.rs

```diff
@@ -146,13 +146,13 @@ fn integration_success() {
     let count = addr_count.lock().unwrap();
     assert_eq!(*count, my_addrs_count);
 
-    // `resolve_count` is not guaranteed to always be 1
-    // or `my_addrs_count`. If `my_addrs_count` > 1, these
-    // addrs could be resolved in a single message from
-    // the daemon, or in separate messages.
-    let count = resolve_count.lock().unwrap();
-    assert!(*count >= 1);
-    assert!(*count <= my_addrs_count);
```

Comment:
Exactly.. Since the cache_flush bit will trigger a new ServiceResolved - and cache_flush bit is being set by the client, the number in `resolve_count` would be higher now. So I introduced the HashSet.